### PR TITLE
feature(zjow): add torch_save method in case of disk being full

### PR DESCRIPTION
**Pull request by zjowowen**

### CODE REVIEW

File: ding/utils/file_helper.py [hunk 1]
```diff
@@ -295,6 +297,32 @@ def read_file(path: str, fs_type: Union[None, str] = None, use_lock: bool = Fals
...
+        - data (:obj:`object`): The data to save
+        - path (:obj:`str`): The path of file to save to
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # save the data to the temporary directory
+        tmp_path = f"{tmpdir}/data.pt"
+        torch.save(data, tmp_path)
+
+        # move the file to the desired path
+        while True:
```
Comment (PaParaZz1) [inline ding/utils/file_helper.py:314]: add a new branch to retry fixed times and ignore disk error, defaults to this branch

File: ding/utils/file_helper.py [hunk 2]
```diff
@@ -295,6 +297,32 @@ def read_file(path: str, fs_type: Union[None, str] = None, use_lock: bool = Fals
...
+                break  # if successful, break out of the loop
+            except OSError as e:
+                if e.errno == 28:  # "No space left on device" error
+                    print("Disk full, waiting for space to be released...")
+                    time.sleep(60)  # wait for a minute and try again
+                else:
+                    raise e  # re-raise any other IOError exceptions
+
+
 def save_file(path: str, data: object, fs_type: Union[None, str] = None, use_lock: bool = False) -> None:
```
Comment (PaParaZz1) [inline ding/utils/file_helper.py:326]: add a argument to control wait or ignore disk full error

File: ding/utils/file_helper.py [hunk 3]
```diff
@@ -295,6 +297,32 @@ def read_file(path: str, fs_type: Union[None, str] = None, use_lock: bool = Fals
...
+        # save the data to the temporary directory
+        tmp_path = f"{tmpdir}/data.pt"
+        torch.save(data, tmp_path)
+
+        # move the file to the desired path
+        while True:
+            try:
+                shutil.move(tmp_path, path)
+                break  # if successful, break out of the loop
+            except OSError as e:
```
Comment (PaParaZz1) [inline ding/utils/file_helper.py:318]: add unittest for this file and error

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: ding/utils/file_helper.py (hunk 1 above)
Comment (zjowowen) [reply ding/utils/file_helper.py]: fixed.

File: ding/utils/file_helper.py (hunk 2 above)
Comment (zjowowen) [reply ding/utils/file_helper.py]: fixed.

File: ding/utils/file_helper.py (hunk 3 above)
Comment (zjowowen) [reply ding/utils/file_helper.py]: fixed.

**Pull request closed without merging**